Amazon_Interview_Questions/Easy/Diameter_Of__Binary_Tree.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+
+Problem Link: https://leetcode.com/problems/diameter-of-binary-tree/
+
+*/
+
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
+ * };
+ */
+
+int ma;
+class Solution {
+    int func(TreeNode *root){
+        if(!root)
+         return 0;
+        int x = func(root -> left);
+        int y = func(root -> right);
+        ma = max(ma, x+y+1);
+        return (max(x,y)+1);
+    }
+public:
+    int diameterOfBinaryTree(TreeNode* root) {
+        ma = INT_MIN;
+        int x = func(root);
+        return ma-1;
+    }
+};
```

Comment:
My 10 months old solution!
Always use meaningful variable names :)
I find it hard to understand what is `int ma`
![image](https://user-images.githubusercontent.com/46818757/118457919-dd4eec80-b717-11eb-96d8-8144ef5145ad.png)

Reply:
@SaumyaSingh1 Okay, I will use meaningful variable names.